share/shwap/proof.go

```diff
@@ -49,64 +46,141 @@ func (p *Proof) RowRootProof() *merkle.Proof {
 
 // VerifyInclusion verifies the inclusion of the shares to the data root.
 func (p *Proof) VerifyInclusion(shares []libshare.Share, namespace libshare.Namespace, dataRoot []byte) error {
-	outside, err := share.IsOutsideRange(namespace, p.root, p.root)
-	if err != nil {
-		return err
+	nth := nmt.NewNmtHasher(
+		share.NewSHA256Hasher(),
+		nmt_ns.ID(namespace.ID()).Size(),
+		p.shareProof.IsMaxNamespaceIDIgnored(),
+	)
+
+	nid := namespace.ID()
+	leaves := libshare.ToBytes(shares)
+
+	// If the proof is empty and no leaves exist, it's a valid proof
+	if p.shareProof.IsEmptyProof() && len(leaves) == 0 {
+		return nil
 	}
```

Comment:
empty proof is not valid proof. the proof is empty, then we must throw an error back to the user. There is no sense for the user to request a range of [0;0]. Could you please double-check 
`func (rngdata *RangeNamespaceData) VerifyShares`. There are conditions for checking Empty and AbsenseProof. I'd propose making these checks inside `VerifyInclusion` under a `ValidateBasic` method. Requesting a range user must exactly know the coordinates of this range, so if he missed smth we have to notify him IMO.

cc @walldiss @Wondertan